showtime-xyz/showtime-frontend
# feat: handling profile 404/error/loading & bug fix

# Why

### profile page has issues: 

- we are not handling not found user cases.
- loading/error has some bug.

### profile page missing social link.


### Setting page: 
We don't have a clear cache feature now: 
1. sometimes when the amount of data in the simulator is too much, if the cache is not clear, it will be stuck.
2. sometimes you don't know if it's a cache issue.

# How

- use the correct `loading` state for `useSWR`.
- remove the hack code, and move the `loading`/`block`/`error` state component to `TabScene`, it should not appear in `rowRenderer`.
- add the social-link feature.
- add the clear-cache feature(only support native), at present, it will only be displayed in `dev`, and we will open it to users when we are sure that we need to.


# Test Plan

- Check the profile page if it works when 404/error/loading.
- Check the profile social link if it works.
